### PR TITLE
support sqlite, mysql and pg out of the box

### PR DESCRIPTION
**Pull request by mitjade**
The idea here is to provide easy transition form `sqlite` to ether `mysql` or `pg`, when you want to use this kit on a real project.

To switch, you need to:
-  set `DB_TYPE ` in `config/secrets.js` form `sqlite` to ether `mysql` or `pg`.
- set `DB_USER `, `DB_PASSWORD ` and `DB_DATABASE ` in `.env`

What should this PR solve:
- no need to override `knexdata.js` or any `databases/seeds` files, witch should make `up-merging` easier
- no need do install new packages like `mysql2` or `pg` and add them to `tools/webpackAppConfig.js`
- have all `env` specific settings in one place -> `.env`
- resolves different strategies needed to `truncate` tables: `common/db` -> `truncateTables`

### CODE REVIEW

**Comment by mitjade** [comment]
I have now projects using this kit, that are running on `mysql` or `pg`. This would resolve merge conflicts on a few files and make transition easier when switching to `mysql` or `pg`.

**Comment by mitjade** [comment]
Later we could provide `docker` setup for this two databases.

For example `docker-compose.pg.yml`:
```
version: '3'
services:
  postgres:
    image: postgres
    environment:
      POSTGRES_PASSWORD: password
    volumes:
      - ${APP_DIR}/data:/var/lib/postgresql/data
    ports:
      - 5432:5432
```

File: .env [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # Misc
 APP_DIR=/usr/src/app
 
+# Db
```
Comment (vlasenko) [inline .env]: @mitjade Could you use `DB` instead of `Db` everywhere?

**Comment by vlasenko** [comment]
@mitjade I think we don't need `config/secrets.js` anymore, as all our secrets are now in `.env`, but this is not responsibility of this PR. Otherwise everything looks good to me, great work, thanks Mitja!

**Comment by mairh** [comment]
The reason for having secret.js is to import `dovenv` in one place rather in each and every file where we are using `process.env.xxxx`. However, I guess it could also work if we just import `dotenv` in server.js?

**Comment by vlasenko** [comment]
@mairh Yes, it should work if we import `dotenv` in `server.js`

File: .env (hunk 1 above)
Comment (mitjade) [reply .env]: done

**Comment by mitjade** [comment]
I agree regarding `secrets.js` not being needed. We just need to find a place for `GOOGLE_TRACKING_ID`, since it will not work if in `.env`, maybe just move it back to `app.json`?

**Comment by vlasenko** [comment]
@mitjade How about moving all properties from `app.json` to `config/app.js` ?

**Comment by mitjade** [comment]
We can try. Should I merge this and we can continue settings optimisations in a different PR?

**Comment by vlasenko** [comment]
@mitjade Sure, lets merge this PR first

**Pull request merged**